### ISSUE TITLE
Image detection precision

### ISSUE THREAD

**Issue by pseudomasq** [Nain57/Smart-AutoClicker#130]
I posted a reply in a different post about this but it is marked as question. Basically, it looks like the difference in color is not considered as different. I have this condition:
![image](https://user-images.githubusercontent.com/76476344/168591239-22b7b630-c03c-4eda-bb1b-e72d47c16df2.png)

that matches with this:
![image](https://user-images.githubusercontent.com/76476344/168591306-894cd911-0ad5-4efb-824c-6e4514edc762.png)

Precision setting is all the way to 1200. 0% tolerance will not even match the actual target so I have to set it to 1% and this issue starts to happen.

**Comment by Nain57** [comment]
Hi and thank you for your feedback.
It seems like a still have issues with the new detection algorithm. The pattern matching from OpenCv can be a bit complicated.

I will investigate this and try to find a solution.

**Comment by Nain57** [comment]
Hi and thank you for your feedback.
It seems like a still have issues with the new detection algorithm. The pattern matching from OpenCv can be a bit complicated.

I will investigate this and try to find a solution.

**Comment by Nain57** [comment]
I have worked at lot on this issue and I'm proud to say that I might have found a working solution !

<p align="center">
  <img src="https://user-images.githubusercontent.com/8599025/169697742-28292951-c85f-4bc4-9d97-c3b41c14e469.jpg" width="200">
  <img src="https://user-images.githubusercontent.com/8599025/169697847-fcb90c71-e474-4e87-a2e4-84b900a4a8e5.jpg" width="200">
  <img src="https://user-images.githubusercontent.com/8599025/169697850-7a03bab8-310a-4d46-94bf-cad85d75f13d.jpg" width="200">
</p>

The bottom picture is always the same, and we are trying to detect it on the whole screen. The top picture changes.
* On the left screenshot, the top picture have a small black filter on it. We detect the correct bottom button.
* On the central screenshot, the top picture have a big black filter on it (almost invisible). We detect the correct bottom button.
* On the right screenshot, the top picture is the same as the bottom one. We detect the top one because we take the first fulfilled detection.

This will be part of the next update that should be released very soon !

**Comment by Nain57** [comment]
I have worked at lot on this issue and I'm proud to say that I might have found a working solution !

<p align="center">
  <img src="https://user-images.githubusercontent.com/8599025/169697742-28292951-c85f-4bc4-9d97-c3b41c14e469.jpg" width="200">
  <img src="https://user-images.githubusercontent.com/8599025/169697847-fcb90c71-e474-4e87-a2e4-84b900a4a8e5.jpg" width="200">
  <img src="https://user-images.githubusercontent.com/8599025/169697850-7a03bab8-310a-4d46-94bf-cad85d75f13d.jpg" width="200">
</p>

The bottom picture is always the same, and we are trying to detect it on the whole screen. The top picture changes.
* On the left screenshot, the top picture have a small black filter on it. We detect the correct bottom button.
* On the central screenshot, the top picture have a big black filter on it (almost invisible). We detect the correct bottom button.
* On the right screenshot, the top picture is the same as the bottom one. We detect the top one because we take the first fulfilled detection.

This will be part of the next update that should be released very soon !

**Comment by Nain57** [comment]
Hi !
Is this issue solved for you on the versions 1.5+ ?

**Comment by pseudomasq** [comment]
Yes. Great work. Wanted to test with the latest release but I didn't see any apk.

Also, I'm wondering if this issue could be made a feature. There could be situations where a color blind detection will be useful like detecting close buttons with different background color in ads.

**Comment by Nain57** [comment]
I forgot to add the apk for the 1.5.1, but the detection is identical between 1.5.0 and 1.51
